json/src/main/scala/com/gu/contentapi/json/Serialization.scala

```diff
@@ -69,8 +71,25 @@ object Serialization {
     case JField("safeEmbedCode", JString(s)) => JField("safeEmbedCode", JBool(s.toBoolean))
   }
 
-  def thriftEnum2JString[A <: ThriftEnum]: PartialFunction[Any, JString] = {
-    case a: ThriftEnum => JString(pascalCaseToHyphenated(a.name))
+  def thriftEnum2JString[A <: ThriftEnum](implicit classTag: ClassTag[A]): PartialFunction[Any, JString] = {
+    case a if classTag.runtimeClass.isInstance(a) => JString(pascalCaseToHyphenated(a.asInstanceOf[A].name))
+  }
```

Comment:
Do you mind explaining what's going on here please? What was happening before? Where is the implicit parameter coming from?